### ISSUE TITLE
Connection error when calling kubernetes-api-server (since 1.1.0-rc.0)

### ISSUE THREAD

**Issue by thomschke** [istio/istio#12121]
**Describe the bug**
Since `v1.1.0-rc.0` all requests agains the api-server `kubernetes.default.svc.cluster.local` like:

```
CA_CERT=/var/run/secrets/kubernetes.io/serviceaccount/ca.crt
TOKEN=$(cat /var/run/secrets/kubernetes.io/serviceaccount/token)
curl -v --cacert $CA_CERT -H "Authorization: Bearer $TOKEN" "https://kubernetes.default/api/v1/"

```
crashes during ssl handshake like:
```
* TCP_NODELAY set
* Connected to kubernetes.default (10.143.0.1) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* Cipher selection: ALL:!EXPORT:!EXPORT40:!EXPORT56:!aNULL:!LOW:!RC4:@STRENGTH
* successfully set certificate verify locations:
*   CAfile: /var/run/secrets/kubernetes.io/serviceaccount/ca.crt
  CApath: /etc/ssl/certs
* TLSv1.2 (OUT), TLS header, Certificate Status (22):
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* Unknown SSL protocol error in connection to kubernetes.default:443
* Curl_http_done: called premature == 1
* stopped the pause stream!
* Closing connection 0
curl: (35) Unknown SSL protocol error in connection to kubernetes.default:443
```
There's no server response at all so I think it's an connection problem (rather than ssl problem).
BTW: It's works perfect from inside pods without any istio sidecar.

I installed istio with Helm/Tiller with identical values.
This issue occurs only from the release `1.1.0-rc.0` and higher.

**Expected behavior**
```
HTTP/2 200
{
  "kind": "APIResourceList",
  "groupVersion": "v1",
  "resources": [
...
```
**Version**
```
version.BuildInfo{Version:"1.1.0-rc.0", GitRevision:"3632361b439ae27962f0af472064d452fc63273b", User:"root", Host:"9efe8d69-34c2-11e9-b805-0a580a2c0506", GolangVersion:"go1.10.4", DockerHub:"docker.io/istio", BuildStatus:"Clean", GitTag:"1.1.0-snapshot.6-26-g3632361"}
```
```
Client Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.3", GitCommit:"721bfa751924da8d1680787490c54b9179b1fed0", GitTreeState:"clean", BuildDate:"2019-02-04T04:49:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"darwin/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.6-gke.3", GitCommit:"04ad69a117f331df6272a343b5d8f9e2aee5ab0c", GitTreeState:"clean", BuildDate:"2019-01-10T00:39:15Z", GoVersion:"go1.10.3b4", Compiler:"gc", Platform:"linux/amd64"}
```

**Installation**
-  [install with helm and tiller](https://preliminary.istio.io/docs/setup/kubernetes/helm-install/#option-2-install-with-helm-and-tiller-via-helm-install)

**Environment**
- Client: MacOS 10.13.6
- Server: GKE 1.11.6-gke.3

**Comment by thomschke** [comment]
Same problem with new pre-release `1.1.0-rc.1` :-(

- I deploy istio with Helm/Tiller and identical values.
- I run `curl` based on multiple images (debian, ...) 

The k8s-api-query runs without any ssl problems under `1.1.0-snapshot.6` and older.

**Comment by thomschke** [comment]
due to #12157

**Comment by sdake** [comment]
@thomschke thanks for finding this problem.  I've been chasing after this for a week!

**Comment by cgkades** [comment]
Is this fixed? I'm having the same/similar issue in 1.1.1

root@dummy-69f5c66886-rhbkw:/# curl https://kubernetes.default.svc.cluster.local
curl: (35) Unknown SSL protocol error in connection to kubernetes.default.svc.cluster.local:443
root@dummy-69f5c66886-rhbkw:/# curl https://kubernetes
curl: (35) Unknown SSL protocol error in connection to kubernetes:443
